### DISCUSSION TITLE
Web Component visible only after window resize

### DISCUSSION THREAD

**Discussion by hkrawczyk** [vercel/next.js#48390] [Help]
### Summary

I've used an external web component that imitates page flipping.
The component works perfectly fine in pure HTML (images have the right size and are visible).
For some reason when used in Next, the component is rendered properly (takes the right sizes) but the images are not visible until I'll change the size of the window. 

Is there any workaround to simulate window resize to force the component to show up?

The Web Component is dynamically imported and is not rendered on the server.

**Comment by 808vita** [comment]
can you post code snippets ?

**Reply by hkrawczyk** [reply]
@808vita, actually it is quite challenging to do it on some codepen, but here you have a repository with CLI generated next + one CSS class & web component.

https://github.com/hkrawczyk/nextjs-web-component-error

Could you please suggest how to re-render the web-component so could appear? 
It is blank until I resize the window.

**Reply by 808vita** [reply]
tried few things , faced the same issue with that repo .

seems like vue based ?
Are you sure this can run well with react / next js ?

and also the link you shared , linear (?) not able to access it .

**Reply by hkrawczyk** [reply]
@808vita idk why it is linear link -  it is auto-fill.

WebComponents are built in a way that make them reusable across all frameworks. Is there any way to force re-render?

**Reply by 808vita** [reply]
Usually re-render can triggered by resetting a state (but did not work out, tried it out ). 
Also tried out with useEffect  , conditionals , import callback  , but nope .

There is something missing, the network requests are successful & visible in network inspection.
The final component is placed inside a shadow dom , that's the lead I guess ?
![image](https://user-images.githubusercontent.com/97225946/233168669-d163ae91-a8e2-4f02-bddb-52c215459fe6.png)


best bet - you should probably refer to that WebComponents (?) documentation , if it exists.

**Reply by 808vita** [reply]
One thing what I have not tired but might work is re-rendering the **whole page** instead of the just the component. 
reason - shadow dom will also get re-rendered in the process . (but not sure though)

**Reply by 808vita** [reply]
![image](https://user-images.githubusercontent.com/97225946/233172562-ea4c0ec1-dfff-48cb-a307-ea2ca8590e09.png)
![image](https://user-images.githubusercontent.com/97225946/233172491-d426c175-80b8-41d6-a1ce-8e7fadc534b9.png)


okay after some more experimentation , react not able to re-render what ever is in that shadow root.

**Reply by hkrawczyk** [reply]
@808vita omg that are great finding!
Sadly there is no documentation for this component. 
In reality, it seems like just resizing some wrapped div could help but I also couldn't find ANY solution.
Really weird... im open for doing any dirty releads just to make it work...

**Reply by 808vita** [reply]
Actually there is way .
Since you already did mention it works on pure html , make it in pure html and place that file in public folder.
Then access the url and it will be served.

**Reply by hkrawczyk** [reply]
> Actually there is way . Since you already did mention it works on pure html , make it in pure html and place that file in public folder. Then access the url and it will be served.

wait, like just iframe it?
overall great idea, just not sure how to implement it to make it scalable as it is in webcomponent!

**Reply by 808vita** [reply]
you can serve it as separate static page or inside the iframe .
If you serve it with iframe then it will work along side nextjs flow.
Scalable here would be ? did you mean showing a different books based on props ? multiple books ?

Attach the required url as query , like "/?bookurl=bookurl" 
(few things note - make sure to encode the url , attach query to that static page url & getting query url data will be done within that static page )

another thing is that there are restrictions on how to access stuff with iframe , do read the docs . But I feel url query params is the simpler and less complex option 

For multiple books -> just create multiple instance of iframe & load that page .

**Reply by 808vita** [reply]
and I think I have found the webComponent package ? vue based .

https://ts1.github.io/flipbook-vue/#1

https://github.com/ts1/flipbook-vue

**Reply by hkrawczyk** [reply]
> you can serve it as separate static page or inside the iframe . If you serve it with iframe then it will work along side nextjs flow. Scalable here would be ? did you mean showing a different books based on props ? multiple books ?
> 
> Attach the required url as query , like "/?bookurl=bookurl" (few things note - make sure to encode the url , attach query to that static page url & getting query url data will be done within that static page )
> 
> another thing is that there are restrictions on how to access stuff with iframe , do read the docs . But I feel url query params is the simpler and less complex option
> 
> For multiple books -> just create multiple instance of iframe & load that page .

by scaling I meant that it is responsive, thank you for the idea!

**Reply by hkrawczyk** [reply]
> and I think I have found the webComponent package ? vue based .
> 
> https://ts1.github.io/flipbook-vue/#1
> 
> https://github.com/ts1/flipbook-vue

thats right - thats the package. it works perfectly fine in vue and as web component, but i had to built it to component on my own

**Reply by 808vita** [reply]
you can use regular css for scaling , style it normally like a div
